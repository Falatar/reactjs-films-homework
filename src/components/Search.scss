.search {
    form {
        position: relative;
        width: 300px;
        margin-right: 6vw;
        margin-top: 1vh;
    }
    input {
        width: 100%;
        height: 42px;
        padding-left: 10px;
        border: 2px solid white;
        border-radius: 10px;
        outline: none;
        background-color: transparent;
        color: white;
    }
    input::-webkit-input-placeholder { color: white; }
    input::-moz-placeholder { color: white; }
    button {
        position: absolute; 
        top: 0;
        right: 0px;
        width: 42px;
        height: 42px;
        border: none;
        background: transparent;
        border-radius: 10px;
        cursor: pointer;
        color: white;
        font-size: 2em;
    }
}