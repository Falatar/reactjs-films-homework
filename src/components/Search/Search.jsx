import React from 'react';
import style from './Search.scss';

function Search() {
  return (
    <div className={style.Search}>
      <form className={style.SearchForm}>
        <input type="text" placeholder="Search here..." className={style.SearchText} />
        <button type="submit" id="push" className={style.SearchButton}>âŒ•</button>
      </form>
    </div>
  );
}

export default Search;
