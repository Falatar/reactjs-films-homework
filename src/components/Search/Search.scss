.Search {
    .SearchForm {
        position: relative;
        width: 300px;
        margin-right: 6vw;
        margin-top: 1vh;
    }
    .SearchText {
        font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
        width: 100%;
        height: 42px;
        padding-left: 10px;
        border: 2px solid white;
        border-radius: 10px;
        outline: none;
        background-color: transparent;
        color: white;
        outline:none;
    }
    .SearchText::-webkit-input-placeholder { color: white; }
    .SearchText::-moz-placeholder { color: white; }
    .SearchButton {
        position: absolute; 
        top: 0;
        right: 0px;
        width: 42px;
        height: 42px;
        border: none;
        background: transparent;
        border-radius: 10px;
        cursor: pointer;
        color: white;
        font-size: 2em;
    }
}