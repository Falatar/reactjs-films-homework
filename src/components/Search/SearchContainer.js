import { connect } from 'react-redux';
import { saveSearchStr } from '../../modules/filmListModule/filmListModuleActions';
import {
  getActualMode, getActualGenre, getCurrentPage, getActiveView, getSearchStr,
} from '../../modules/filmListModule/filmListModuleSelector';
import Search from './Search';

export const mapStateToProps = (store) => ({
  activeMode: getActualMode(store),
  activeGenre: getActualGenre(store),
  actualPage: getCurrentPage(store),
  activeView: getActiveView(store),
  searchStr: getSearchStr(store),
});

export const mapDispatchToProps = (dispatch) => ({
  saveSearchStr: (searchString) => dispatch(saveSearchStr(searchString)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Search);
