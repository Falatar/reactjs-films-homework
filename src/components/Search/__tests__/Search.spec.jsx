import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../Search';

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');

  return {
    __esModule: true,
    ...originalModule,
    Link: 'Link',
    useParams: jest.fn(),
    useHistory: jest.fn(),
  };
});

describe('Search', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Search />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('updateInputValue works correctly', () => {
    const component = renderer.create(<Search />);
    const testEvent = {
      target: {
        value: 29052020,
      },
    };
    component.getInstance().updateInputValue(testEvent);
    expect(component.getInstance().state.value).toEqual(29052020);
  });

  it('enableScan works correctly', () => {
    const search = jest.fn();
    const component = renderer.create(<Search search={search} />);
    component.getInstance().state.value = 'test';
    component.getInstance().enableScan();
    expect(search).toHaveBeenCalled();
  });

  it('enableScan works correctly with default state', () => {
    const search = jest.fn();
    const component = renderer.create(<Search search={search} />);
    component.getInstance().state.value = '';
    component.getInstance().enableScan();
    expect(search).not.toHaveBeenCalled();
  });

  it('input change works correctly', () => {
    const upd = jest.fn();
    const component = renderer.create(
      <Search />,
    );
    component.getInstance().updateInputValue = upd;
    const input = component.root.findByType('input');
    input.props.onChange();
    expect(upd).toHaveBeenCalled();
  });

  it('default function works correctly', () => {
    const component = renderer.create(<Search search={undefined} />);
    const result = component.getInstance().props.search();
    expect(result).toEqual('can\'t find search function');
  });
});
