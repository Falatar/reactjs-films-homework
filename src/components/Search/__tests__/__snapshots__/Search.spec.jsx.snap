// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="Search"
>
  <form
    className="SearchForm"
  >
    <input
      className="SearchText"
      placeholder="Search here..."
      type="text"
    />
    <button
      className="SearchButton"
      id="push"
      type="submit"
    >
      ⌕
    </button>
  </form>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="Search"
>
  <form
    className="SearchForm"
  >
    <input
      className="SearchText"
      placeholder="Search here..."
      type="text"
    />
    <button
      className="SearchButton"
      id="push"
      type="submit"
    >
      ⌕
    </button>
  </form>
</div>
`;
